helm/cluster-api-ipam-provider-in-cluster/templates/pools/job.yaml

```diff
@@ -57,12 +57,20 @@ spec:
         volumeMounts:
         - name: in-cluster-ip-pool
           mountPath: /data/pool.yaml
+          subPath: pool.yaml
```

Comment:
Do we need `pool.yaml` in both `subPath` and `mountPath`?  

`kubectl apply -f /data/` is the command we run in the container. Why not mount all files under `/data` here?

Reply:
I've tried and without the subPath it's actually mounted as a directory called `pool.yaml/` in which there is a file called `pool.yaml`

```
k exec -ti create-in-cluster-ip-pool-svc7x -- cat /data/pool.yaml                                                                                                                        ○ gipam-admin@gipam
Defaulted container "create-in-cluster-ip-pool" out of: create-in-cluster-ip-pool, wait-crds (init)
cat: read error: Is a directory
command terminated with exit code 1
~/workspace/cluster-api-ipam-provider-in-cluster-app add-subPath *3 !2 ?1 λ k exec -ti create-in-cluster-ip-pool-svc7x -- ls /data/pool.yaml                                                                                                                         ○ gipam-admin@gipam
Defaulted container "create-in-cluster-ip-pool" out of: create-in-cluster-ip-pool, wait-crds (init)
pool.yaml
~/workspace/cluster-api-ipam-provider-in-cluster-app add-subPath *3 !2 ?1 λ k exec -ti create-in-cluster-ip-pool-svc7x -- cat /data/pool.yaml/pool.yaml                                                                                                              ○ gipam-admin@gipam
Defaulted container "create-in-cluster-ip-pool" out of: create-in-cluster-ip-pool, wait-crds (init)
apiVersion: ipam.cluster.x-k8s.io/v1alpha1
kind: InClusterIPPool
metadata:
  name: wc-cp-ips
spec:
  gateway: 0.0.0.0
  prefix: 24
  start: 10.10.222.222
  end: 10.10.222.229
```

I did the same thing here as we do in our `crd-install` hook.

`gjiri` MC (but any vsphere cluster will do):
```
helm --kubeconfig ~/.kube/own/gjiri.kubeconfig get hooks cluster-api-provider-vsphere -n giantswarm | grep -A6 volumeMounts:
        volumeMounts:
        - name: vsphereclusteridentities
          mountPath: /data/vsphereclusteridentities.yaml
          subPath: vsphereclusteridentities.yaml
        - name: vsphereclusters
          mountPath: /data/vsphereclusters.yaml
          subPath: vsphereclusters.yaml
```